File: src/main/java/calebzhou/rdi/microservice/model/json/GeoLocation.java
package calebzhou.rdi.microservice.model.json;

import calebzhou.rdi.microservice.utils.RdiSerializer;

import java.io.Serializable;

/**
 * Created by calebzhou on 2022-09-20,16:16.
 */
public class GeoLocation implements Serializable {
	public double latitude;
	public double longitude;

	public GeoLocation(double latitude, double longitude) {
		this.latitude = latitude;
		this.longitude = longitude;
	}
	@Override
	public String toString() {
		return RdiSerializer.GSON.toJson(this);
	}
	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		GeoLocation that = (GeoLocation) o;
		return that.latitude == this.latitude && that.longitude == this.longitude;
	}

}

File: src/main/java/calebzhou/rdi/microservice/model/GeoLocation.kt
package calebzhou.rdi.microservice.model

import java.sql.Timestamp

/**
 * Created by calebzhou on 2022-10-03,21:09.
 */

/**
 *  地理天气
 */
//经度纬度
class GeoLocation(var latitude:Double,var longitude:Double)
//RDI地址位置
class RdiGeoLocation(var nation: String, var province: String, var city: String,
                     var district: String, var isp: String, var location: GeoLocation)
//ip2region开原地址库的信息
class Ip2RegionData(val nation: String, val province: String, val city: String, val isp: String)
